Remove unused PlatformMouseEvent::modifierFlags() and m_modifierFlags

ddkilzer opened the pull request:
#### c559ef21e24775338a1eb865f46d10d60f0bb953
<pre>
Remove unused PlatformMouseEvent::modifierFlags() and m_modifierFlags
<a href="https://bugs.webkit.org/show_bug.cgi?id=245108">https://bugs.webkit.org/show_bug.cgi?id=245108</a>
&lt;<a href="https://rdar.apple.com/99850228">rdar://99850228</a>&gt;

Reviewed by Darin Adler.

PlatformMouseEvent::modifierFlags() was last called from EventHandler to
pass raw modifier flags to Chrome::mouseDidMoveOverElement(), which was
removed in 264519@main for Bug 257175.  Since then, no code reads the
field.

Modifier information is already available through the base class via
PlatformEvent::modifiers() as an OptionSet&lt;PlatformEvent::Modifier&gt;.

* Source/WebCore/platform/PlatformMouseEvent.h:
(WebCore::PlatformMouseEvent::modifierFlags const): Delete.
(WebCore::PlatformMouseEvent::m_modifierFlags): Delete.
* Source/WebCore/platform/mac/PlatformEventFactoryMac.mm:
(WebCore::PlatformMouseEventBuilder::PlatformMouseEventBuilder):
* Source/WebCore/platform/win/PlatformMouseEventWin.cpp:
(WebCore::PlatformMouseEvent::PlatformMouseEvent):
* Source/WebKit/Shared/WebEventConversion.cpp:
(WebKit::WebKit2PlatformMouseEvent::WebKit2PlatformMouseEvent):
- Remove unused modifierFlags() and m_modifierFlags.

* Tools/TestWebKitAPI/Tests/mac/MenuTypesForMouseEvents.mm:
(TestWebKitAPI::buildAndPerformTest):
- Replace the modifierFlags() assertion with a single EXPECT_EQ against
  pme.modifiers(), using modifiersForModifierFlags() to convert the
  NSEventModifierFlags parameter to an OptionSet&lt;PlatformEvent::Modifier&gt;.

Canonical link: <a href="https://commits.webkit.org/310567@main">https://commits.webkit.org/310567@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/e4dd3baa0614fe2f80c9d0f9dc1ab4d9a2c368f9

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows | Apple Internal |
| ----- | ---------------------- | ------- |  ----- |  --------- | ------ |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/154197 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/159/builds/27454 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/168/builds/20615 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/162951 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/107665 "Built successfully") | ⏳ 🛠 ios-apple 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/156070 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/155/builds/27587 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/156/builds/27305 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/119255 "Passed tests") | [✅ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/84303 "Passed tests") | ⏳ 🛠 mac-apple 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/157156 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/162/builds/21518 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/138477 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/99951 "Passed tests") | | [✅ 🛠 vision-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/9e3e4f99-fedf-4da3-8a91-4b42e965e839/0b357289-62a6-4c45-ae2b-45d76849a2bd) 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/154/builds/20606 "Passed tests") | [✅ 🧪 api-mac-debug](https://ews-build.webkit.org/#/builders/165/builds/18603 "Passed tests") | [✅ 🛠 gtk3-libwebrtc](https://ews-build.webkit.org/#/builders/173/builds/10783 "Built successfully") | | 
| | [❌ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/130268 "Found 1 new API test failure: TestWebKitAPI.WKDownload.OriginatingFrameHostWhenDownloadComesFromClientInputNavigation (failure)") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/167/builds/16322 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/165423 "Built successfully") | | 
| | [✅ 🛠 ios-safer-cpp](https://ews-build.webkit.org/#/builders/174/builds/8632 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/169/builds/17931 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/127350 "Passed tests") | | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/153/builds/27001 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/161/builds/22645 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/127495 "Passed tests") | | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/160/builds/26925 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/138115 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/83518 "Built successfully") | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/23555 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/164/builds/22383 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/170/builds/14907 "Passed tests") | | | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/158/builds/26615 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/90718 "Built successfully") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/157/builds/26196 "Built successfully") | | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/163/builds/26427 "Built successfully") | | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/152/builds/26268 "Built successfully") | | | | 
<!--EWS-Status-Bubble-End-->

darinadler commented on Source/WebKit/Shared/WebEventConversion.cpp:
```diff
@@ -305,18 +312,6 @@ class WebKit2PlatformMouseEvent : public WebCore::PlatformMouseEvent {
         m_syntheticClickType = static_cast<WebCore::SyntheticClickType>(webEvent.syntheticClickType());
 #endif
-        m_modifierFlags = 0;
-        if (webEvent.shiftKey())
-            m_modifierFlags |= static_cast<unsigned>(WebEventModifier::ShiftKey);
-        if (webEvent.controlKey())
-            m_modifierFlags |= static_cast<unsigned>(WebEventModifier::ControlKey);
-        if (webEvent.altKey())
-            m_modifierFlags |= static_cast<unsigned>(WebEventModifier::AltKey);
-        if (webEvent.metaKey())
-            m_modifierFlags |= static_cast<unsigned>(WebEventModifier::MetaKey);
```
How did you determine this isn't needed on platforms other than MAC and WIN? Change log says we are removing it but not why that is OK.

ddkilzer replied:
The replacement for this code on Mac is at lines 304-306, which correctly converts `webEvent.modifiers()` to `NSEventModifierFlags` via `toNSEventModifierFlags()`, instead of incorrectly storing raw `OptionSet` enum bits.

The removed code was a regression from [r188192](https://commits.webkit.org/165926@main) ([Bug 171297](https://bugs.webkit.org/show_bug.cgi?id=171297)): it stored raw `WebEventModifier` enum bit values (1, 2, 4, 8, …) into `m_modifierFlags`, whose type (`NSEventModifierFlags`) uses a completely different bitmask.

`mouseModifierFlags()` is only defined as a real accessor on `PLATFORM(MAC)` and `PLATFORM(WIN)`; on all other platforms it returns 0. Since `WebEventFactory::toNSEventModifierFlags()` is a Mac-only function, the `#if PLATFORM(MAC)` guard on the replacement makes it Mac-only here as well.

ddkilzer replied:
Also updated the commit log message.

darinadler replied:
> `mouseModifierFlags()` is only defined as a real accessor on `PLATFORM(MAC)` and `PLATFORM(WIN)`; on all other platforms it returns 0.

Before this change, would `modifierFlags()` have returned a non-zero value in some cases when running on an iPad, for example?

ddkilzer replied:
Possibly.  Need to check what `PLATFORM(MACCATALYST)` has defined when it builds.

Actually, that's not the only use case, either, because of mouse support on iPad for Sidecar.

Won't land until that's cleared up.

ddkilzer replied:
Turns out all of this is dead code now since [264519@main](https://commits.webkit.org/264519@main) for [Bug 257175](http://bugs.webkit.org/show_bug.cgi?id=257175) landed; see PR https://github.com/WebKit/WebKit/pull/14230.

Commit:
Remove unused PlatformMouseEvent::modifierFlags() and m_modifierFlags
https://bugs.webkit.org/show_bug.cgi?id=245108
<rdar://99850228>

Reviewed by Darin Adler.

PlatformMouseEvent::modifierFlags() was last called from EventHandler to
pass raw modifier flags to Chrome::mouseDidMoveOverElement(), which was
removed in 264519@main for Bug 257175.  Since then, no code reads the
field.

Modifier information is already available through the base class via
PlatformEvent::modifiers() as an OptionSet<PlatformEvent::Modifier>.

* Source/WebCore/platform/PlatformMouseEvent.h:
(WebCore::PlatformMouseEvent::modifierFlags const): Delete.
(WebCore::PlatformMouseEvent::m_modifierFlags): Delete.
* Source/WebCore/platform/mac/PlatformEventFactoryMac.mm:
(WebCore::PlatformMouseEventBuilder::PlatformMouseEventBuilder):
* Source/WebCore/platform/win/PlatformMouseEventWin.cpp:
(WebCore::PlatformMouseEvent::PlatformMouseEvent):
* Source/WebKit/Shared/WebEventConversion.cpp:
(WebKit::WebKit2PlatformMouseEvent::WebKit2PlatformMouseEvent):
- Remove unused modifierFlags() and m_modifierFlags.

* Tools/TestWebKitAPI/Tests/mac/MenuTypesForMouseEvents.mm:
(TestWebKitAPI::buildAndPerformTest):
- Replace the modifierFlags() assertion with a single EXPECT_EQ against
  pme.modifiers(), using modifiersForModifierFlags() to convert the
  NSEventModifierFlags parameter to an OptionSet<PlatformEvent::Modifier>.

Canonical link: https://commits.webkit.org/310567@main
